lib/orbit-common/local-storage-source.js

```diff
@@ -59,16 +61,28 @@ var LocalStorageSource = MemorySource.extend({
     }
   },
 
+  getKey: function(path) {
+    return [this.namespace, path[0], path[1]].join(this.delimiter);
+  },
+
+
   /////////////////////////////////////////////////////////////////////////////
   // Internals
   /////////////////////////////////////////////////////////////////////////////
 
-  _saveData: function(forceSave) {
-    if (!this._autosave && !forceSave) {
+  _saveData: function(operation) {
+    if (!this._autosave && !operation) {
       this._isDirty = true;
       return;
     }
-    window.localStorage.setItem(this.namespace, JSON.stringify(this.retrieve()));
+    var obj = this.retrieve([operation.path[0], operation.path[1]]);
+
+    if (operation.op === 'add' || operation.op === 'replace') {
```

Comment:
the following logic is only valid when `path.length === 2` -- all other operations should probably still save the `operation.path.slice(0, 2)` value